### ISSUE TITLE
Model not saving after training 

### ISSUE THREAD

**Issue by majinshaoyuindustry** [ennauata/housegan#18]
Hi everyone, 

I am new to GAN and torch so please bear with my limited knowledge here.

I was trying to train on set E and everything seemed to be fine until before the last iteration of the last epoch ran, it exited without saving the model .pth file. The change I made in the Main.py file are #1 Directories and Names based on my own machine #2 I added `if __name__ ==  '__main__':` between line 20 (from models import...) and line 23 (where the parser starts) in order to have it run on WIN10

Btw there is no problem whatsoever running the pre-trained model. I was trying to train a model for more rooms/bigger graph size, like 20+. 

Start: `PS C:\machinelearning\housegan> python main.py --target_set E --exp_folder exp_example
134626
target samples: defaultdict(<class 'int'>, {3: 4126, 7: 18971, 9: 16395, 6: 16179, 11: 13357, 1: 2138, 12: 9390, 10: 15260, 8: 17591, 4: 7005, 5: 11547, 2: 2667})
5000
target samples: defaultdict(<class 'int'>, {13: 2561, 17: 108, 14: 1323, 18: 53, 15: 599, 16: 241, 20: 24, 24: 12, 19: 39, 23: 8, 22: 17, 21: 8, 34: 1, 32: 1, 26: 2, 25: 2, 30: 1})
[Epoch 0/1] [Batch 0/2104] [D loss: 9.999939] [G loss: 0.085443]
[Epoch 0/1] [Batch 1/2104] [D loss: 9.999902] [G loss: 0.087042]
[Epoch 0/1] [Batch 2/2104] [D loss: 9.999855] [G loss: 0.089001]`

End: `[Epoch 0/1] [Batch 2102/2104] [D loss: -36.034729] [G loss: -988.335083]
[Epoch 0/1] [Batch 2103/2104] [D loss: -32.958317] [G loss: -837.985046]
PS C:\machinelearning\housegan>`

Please advise, thank you!

**Comment by ennauata** [comment]
Hello @majinshaoyuindustry,

I wonder if during training the intermediate checkpoints were saved normally? In the current version, we are saving the model after every X iterations and not every Y epochs. So it may or may not save after the last iteration depending on the length//batch_size of our data set. To save the last iteration, you may need to repeat this line after the outermost loop:
`torch.save(generator.state_dict(), './checkpoints/{}_{}.pth'.format(exp_folder, batches_done))`
